projects/demo/src/modules/components/input-date/examples/5/native-date-transformer.directive.ts

```diff
@@ -0,0 +1,28 @@
+import {Directive} from '@angular/core';
+import {AbstractTuiControlValueTransformer, TuiDay} from '@taiga-ui/cdk';
+import {TUI_DATE_VALUE_TRANSFORMER} from '@taiga-ui/kit';
+
+type From = TuiDay | null;
+
+type To = Date | null;
```

Comment:
What do you think, could we use these types in the future?

```ts
export type TuiDateFrom = TuiDay | null;
export type TuiDateTo = Date | null;
```

and then 

```
export class ExampleDateTransformer extends AbstractTuiControlValueTransformer<
    TuiDateFrom,
    TuiDateTo
> {
    toOrigin(transformedValue: TuiDateFrom): TuiDateFrom {
        return transformedValue && TuiDay.fromLocalNativeDate(transformedValue);
    }

    transformValue(originValue: TuiDateFrom): TuiDateTo {
        return originValue && originValue.toLocalNativeDate();
    }
}
```

Reply:
It is just an example in `demo`-app.
User transforms our `TuiDay` to any desirable format: native Date, monent.js, date-fns etc.

As for us, we are not going to use this type (with native Date) in our source code.
So, i think it is not necessary to introduce these types.